modules/querier/querier_query_range.go

```diff
@@ -121,13 +124,19 @@ func (q *Querier) queryBlock(ctx context.Context, req *tempopb.QueryRangeRequest
 
 	inspectedBytes, spansTotal, _ := eval.Metrics()
 
-	return &tempopb.QueryRangeResponse{
+	response := &tempopb.QueryRangeResponse{
 		Series: queryRangeTraceQLToProto(res, req),
 		Metrics: &tempopb.SearchMetrics{
 			InspectedBytes: inspectedBytes,
 			InspectedSpans: spansTotal,
 		},
-	}, nil
+	}
+
+	if len(res) > int(req.MaxSeries) {
+		response.Status = tempopb.PartialStatus_PARTIAL
```

Comment:
truncate the series instead of marshalling them all and passing them back? or really truncate it before we `queryRangeTraceQLToProto`?

similar to above the evaluator could just truncate it internally and let us know that it had truncated results? it looks like we're already passing max series in when we call Do()? 

maybe we require that parameter at creation time and Do returns if the limit was exceeded?

cc @mdisibio